scripts/setup-new-module

```diff
@@ -0,0 +1,406 @@
+#!/usr/bin/env python
+
+"""
+setup-new-module - Script to set up a new module in the AWS SDK for Java v2
+Usage: ./scripts/setup-new-module -n module-name [-t] [-p parent-dir]
+Options:
+  -n module-name: Name of the new module (required)
+  -t: Set up as a test module (optional, default: false)
+  -p parent-dir: Parent directory for the module (optional, default: root project directory for regular modules, test directory for test modules)
+  -h: Show help
+"""
+
+import os
+import sys
+import argparse
+import re
+import shutil
+from pathlib import Path
+
+
+def parse_arguments():
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(description='Set up a new module in the AWS SDK for Java v2')
+    parser.add_argument('-n', '--name', required=True, help='Name of the new module (required)')
+    parser.add_argument('-t', '--test', action='store_true', help='Set up as a test module (optional, default: false)')
+    parser.add_argument('-p', '--parent-dir', help='Parent directory for the module (optional)')
+    return parser.parse_args()
+
+
+def get_sdk_version(root_dir):
+    """Get the current SDK version from the root pom.xml."""
+    pom_path = os.path.join(root_dir, 'pom.xml')
+    with open(pom_path, 'r') as f:
+        content = f.read()
+    
+    # Find the first version tag
+    match = re.search(r'<version>([^<]+)</version>', content)
+    if match:
+        return match.group(1)
+    else:
+        print("Warning: Could not find SDK version in pom.xml")
+        return "UNKNOWN"
+
+
+def create_module_directory(module_dir):
+    """Create the module directory structure."""
+    print(f"Creating module directory: {module_dir}")
+    os.makedirs(module_dir, exist_ok=True)
+    os.makedirs(os.path.join(module_dir, 'src/main/java'), exist_ok=True)
+    os.makedirs(os.path.join(module_dir, 'src/main/resources'), exist_ok=True)
+    os.makedirs(os.path.join(module_dir, 'src/test/java'), exist_ok=True)
+    os.makedirs(os.path.join(module_dir, 'src/test/resources'), exist_ok=True)
+
+
+def create_pom_xml(module_dir, module_name, sdk_version, is_test_module):
+    """Create a basic pom.xml file for the module."""
+    pom_path = os.path.join(module_dir, 'pom.xml')
+    
+    print(f"Using SDK version: {sdk_version}")
+    
+    with open(pom_path, 'w') as f:
+        f.write(f'''<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
Nit: put the start of text on new line for readability and consistency with other text below